src/libs/type-utils/type-helpers.ts

```diff
@@ -1,4 +1,18 @@
-// This syntax gives us nominal typing
-// Nominal types are not interchangable even if the underlying type is the same
-// A func that takes (a: A, b: B) where A and B are both strings cannot be called like (b,a) with this restriction
+/** Allows for nominal typing
+ * Nominal types are not interchangable even if the underlying type is the same
+ * A func that takes (a: A, b: B) where A and B are both strings cannot be called like (b,a) with this restriction
+ */
 export type NominalType<BaseType, Name extends string> = BaseType & { __typeToken: Name }
+
+/**
+ * Type predicate that will check if variable (obj) is a Fetch Response
+ * @param obj
+ */
+export const isFetchResponse = (obj: unknown): obj is Response => {
```

Comment:
Why do we need something more complicated than `instanceof Response`?

Reply:
I generally favor duck-typing... "if it quacks like a duck, it's a duck".  This follows the "Program to interfaces, not instantiations" design principle.  But I am open to counter-points...

Reply:
If going with duck typing, all `useLoadedData` uses is the `text` method. Why also require `status` and `statusText` properties?

Reply:
This was to compensate for not being able to safely check that text was a method that returned a promise without calling it... status and statusText seemed like a reasonable "surface" to test against.  Duck typing also makes mocking way easier... which also influenced these three props as being what is checked against for the type predicate.